server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -506,6 +506,10 @@ public class ConfigurationManagerImpl extends ManagerBase implements Configurati
     public static final ConfigKey<Boolean> ALLOW_DOMAIN_ADMINS_TO_CREATE_TAGGED_OFFERINGS = new ConfigKey<>(Boolean.class, "allow.domain.admins.to.create.tagged.offerings", "Advanced",
             "false", "Allow domain admins to create offerings with tags.", true, ConfigKey.Scope.Account, null);
 
+    public static final ConfigKey<Long> DELETE_BATCH_QUERY_SIZE = new ConfigKey<>("Advanced", Long.class, "delete.batch.query.size", "0",
+            "Indicates the limit applied while deleting entries in bulk. With this, ACS will run the delete query, applying the limit, as many times as necessary" +
+            " to delete all the entries. Zero (0) means that no limit will be applied, therefore, the query will run once and without limit, keeping the default behavior.", true);
+
```

Comment:
@vishesh92 Add upgrade path to change config name` vm.stats.remove.batch.size` => `delete.batch.query.size`

Reply:
This is within the same minor version. Do we need an upgrade path?

Reply:
ok, then not needed @vishesh92